File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
    queue<int> q;

    // 입력 받기
    int num_commands;
    cin >> num_commands;

    // 명령어 처리
    for (int i = 0; i < num_commands; i++) {
        string command;
        cin >> command;

        if (command == "push") {
            int value;
            cin >> value;
            q.push(value);
        } else if (command == "pop") {
            if (!q.empty()) {
                cout << q.front() << endl;
                q.pop();
            } else {
                cout << -1 << endl;
            }
        } else if (command == "size") {
            cout << q.size() << endl;
        } else if (command == "empty") {
            cout << q.empty() << endl;
        } else if (command == "front") {
            if (!q.empty()) {
                cout << q.front() << endl;
            } else {
                cout << -1 << endl;
            }
        } else if (command == "back") {
            if (!q.empty()) {
                cout << q.back() << endl;
            } else {
                cout << -1 << endl;
            }
        }
    }

    return 0;
}

File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline

q = []

for _ in range(int(input())):
    s = input().strip().split()
    if s[0] == "push" : q.append(s[1])
    if s[0] == "pop" : print(q.pop(0) if q else -1)
    if s[0] == "size" : print(len(q))
    if s[0] == "empty" : print(0 if q else 1)
    if s[0] == "front" : print(q[0] if q else -1)
    if s[0] == "back" : print(q[-1] if q else -1)